addons/pos_self_order/controllers/orders.py

```diff
@@ -68,14 +68,14 @@ def _get_prefixes(self, device_type):
 
         return tracking_prefix, ref_prefix
 
-    def _generate_return_values(self, order, config_id):
+    def _generate_return_values(self, order, config):
         return {
-            'pos.order': order.read(order._load_pos_data_fields(config_id.id), load=False),
-            'res.partner': order.partner_id.read(order.partner_id._load_pos_data_fields(config_id.id), load=False),
-            'pos.order.line': order.lines.read(order._load_pos_data_fields(config_id.id), load=False),
-            'pos.payment': order.payment_ids.read(order.payment_ids._load_pos_data_fields(order.config_id.id), load=False),
-            'pos.payment.method': order.payment_ids.mapped('payment_method_id').read(order.env['pos.payment.method']._load_pos_data_fields(order.config_id.id), load=False),
-            'product.attribute.custom.value':  order.lines.custom_attribute_value_ids.read(order.lines.custom_attribute_value_ids._load_pos_data_fields(config_id.id), load=False),
+            'pos.order': order.read(order._load_pos_data_fields(config), load=False),
```

Comment:
Here we should not do 

```suggestion
            'pos.order': self.env['pos.order']._load_pos_data_read(order, config),
```

?

Same for all